calcom/cal.com
# feat: Overlay Calendar v2 and Troubleshooter v2

## What does this PR do?




- This PR currently only fetches event titles for Google and Office365 Calendar.
- This PR allows users to use Overlay Calendar feature by signing in using their Google Account.
- It also pulls actual event title from the calendar instead of just showing "Busy" for both overlay calendar and troubleshooter.
- This feature upgrade does not need any Google scope changes.
- As per the comment here https://github.com/calcom/cal.com/pull/14575#pullrequestreview-2000344073, I am sticking with the existing Google OAuth Provider.  But since we're using it for both regular and overlay user sign-ups, telling them apart during registration is tricky. So, I couldn't automate Google Calendar setup for overlay users after sign-up. Instead, I'm redirecting them to "/getting-started/connect-calendar" to connect it manually. Let me know if you have any other ideas.

Fixes #12763 
Fixes #12752 



## Type of change


- New feature (non-breaking change which adds functionality)

## How should this be tested?



### Testing Overlay Calendar v2

- Just go to a booking page and toggle the Overlay Calendar switch.
- Choose Continue with Google and sign in with a google id.
- You will be taken to "/getting-started/connect-calendar" page.
- Connect your google calendar.
- You will then be redirected to the booking page.
- You will see that now event titles are being displayed instead of just "Busy" for the calendar busy events.

### Testing Troubleshooter v2

- Just go to the Troubleshooter and make sure that google calendar is connected and the app is checking google calendar for conflicts.
- You will see that now event titles are being displayed instead of just "Busy" for the calendar busy events.


## Mandatory Tasks
